### DISCUSSION TITLE
Outputs outside of problem domain

### DISCUSSION THREAD

**Discussion by Andreja28** [lululxvi/deepxde#1425] [Q&A]
I am working on solving a diffusion PDE with sink using DeepXDE. I have to find concentration of oxygen in a certain depth (x) over time (t). So my input to my network is a pair (x,t), and the output is concentration of oxygen at the given depth in a given time. I also have an analytical solution for the problem.

So basically I have successfully made a model that corresponds to analytical solution. The problem I have is that analytical solution returns a negative concentration for larger times, which is physically not possible. Interperting the physics of the problem I have come to a conclustion that when concentration reaches negative levels it should be interperted as 0 (there is no oxygen left).

My question is how can I tune my network so that the outputs are not negative? In those cases I want outputs to be zero.

I have tried using the `apply_outputs_transform` function (function that maps all negative solutions to zeroes), but it seems that it interferes with the training (the output of the network is slightly off the analytical solution).

Does anyone have some idea on how I am supposed to aproach this?

**Comment by lululxvi** [comment]
`apply_outputs_transform` seems a correct way to do this.
